jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/ResourceService.java

```diff
@@ -678,7 +684,13 @@ private void sendData(Request request, Response response, Callback callback, Htt
             putHeaders(response, content, range.getLength());
             response.setStatus(HttpStatus.PARTIAL_CONTENT_206);
             response.getHeaders().put(HttpHeader.CONTENT_RANGE, range.toHeaderValue(contentLength));
-            Content.copy(new MultiPartByteRanges.PathContentSource(content.getResource().getPath(), range), response, callback);
+            Path path = content.getResource().getPath();
+            Content.Source source;
+            if (path != null)
```

Comment:
Maybe a comment to say why a path is preferred (less copying and async)

Reply:
Done.